### ISSUE TITLE
Potential permissions issue with Homebrew install + Mojave?

### ISSUE THREAD

**Issue by Nickersoft** [Carthage/Carthage#2595]
* carthage install method: [ ] .pkg, [x] homebrew, [ ] source
* `which carthage`: /usr/local/bin/carthage
* `carthage version`: 0.31.0
* `xcodebuild -version`: XCode 10.0, Build Version 10A255
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`?  No
* Are you using `--use-submodules`? No
* Are you using `--cache-builds`? No
* Are you using `--new-resolver`? No

Carthage's install seems to be failing on Homebrew 1.7.6. I'm trying to do a clean install on macOS Mojave with XCode 10.0, and it throws a slew of permissions-related issues trying to install:

```
2018-09-27 20:52:02.872 xcodebuild[11908:59883] [MT] IDELogStore: Failed to open log store at /Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Logs/Build
2018-09-27 20:52:02.872 xcodebuild[11908:59883] [MT] IDELogStore: Failed to open Build log store: Error Domain=IDEFoundationErrorDomain Code=1 "Unable to create log store directory at '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Logs/Build': (513) You don’t have permission to save the file “Build” in the folder “Logs”." UserInfo={NSLocalizedDescription=Unable to create log store directory at '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Logs/Build': (513) You don’t have permission to save the file “Build” in the folder “Logs”.}. User info: {
    NSLocalizedDescription = "Unable to create log store directory at '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Logs/Build': (513) You don\U2019t have permission to save the file \U201cBuild\U201d in the folder \U201cLogs\U201d.";
}.
2018-09-27 20:52:02.960 xcodebuild[11908:59883] Couldn't create intermediates directory: You don’t have permission to save the file “Intermediates.noindex” in the folder “Build”.
2018-09-27 20:52:03.275 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn' (Operation not permitted (1))
2018-09-27 20:52:03.276 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build' (No such file or directory (2))
2018-09-27 20:52:03.276 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex' (No such file or directory (2))
2018-09-27 20:52:03.278 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn' (Operation not permitted (1))
2018-09-27 20:52:03.278 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build' (No such file or directory (2))
2018-09-27 20:52:03.278 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex' (No such file or directory (2))
2018-09-27 20:52:03.278 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex/XCBuildData' (No such file or directory (2))
2018-09-27 20:52:03.278 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex/XCBuildData/PIFCache' (No such file or directory (2))
2018-09-27 20:52:03.278 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex/XCBuildData/PIFCache/project' (No such file or directory (2))
2018-09-27 20:52:03.278 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn' (Operation not permitted (1))
2018-09-27 20:52:03.278 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build' (No such file or directory (2))
2018-09-27 20:52:03.278 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex' (No such file or directory (2))
2018-09-27 20:52:03.278 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn' (Operation not permitted (1))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build' (No such file or directory (2))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex' (No such file or directory (2))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex/XCBuildData' (No such file or directory (2))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex/XCBuildData/PIFCache' (No such file or directory (2))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex/XCBuildData/PIFCache/target' (No such file or directory (2))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn' (Operation not permitted (1))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build' (No such file or directory (2))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex' (No such file or directory (2))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn' (Operation not permitted (1))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build' (No such file or directory (2))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex' (No such file or directory (2))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex/XCBuildData' (No such file or directory (2))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex/XCBuildData/PIFCache' (No such file or directory (2))
2018-09-27 20:52:03.279 XCBBuildService[12347:60610] unable to create directory: '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Build/Intermediates.noindex/XCBuildData/PIFCache/workspace' (No such file or directory (2))
2018-09-27 20:52:03.280 xcodebuild[11908:59883] [MT] IDELogStore: Failed to open log store at /Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Logs/Test
2018-09-27 20:52:03.280 xcodebuild[11908:59883] [MT] IDELogStore: Failed to open Test log store: Error Domain=IDEFoundationErrorDomain Code=1 "Unable to create log store directory at '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Logs/Test': (513) You don’t have permission to save the file “Test” in the folder “Logs”." UserInfo={NSLocalizedDescription=Unable to create log store directory at '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Logs/Test': (513) You don’t have permission to save the file “Test” in the folder “Logs”.}. User info: {
    NSLocalizedDescription = "Unable to create log store directory at '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn/Logs/Test': (513) You don\U2019t have permission to save the file \U201cTest\U201d in the folder \U201cLogs\U201d.";
}.
Build Preparation
Couldn't create workspace arena folder '/Users/tjnickerson/Library/Developer/Xcode/DerivedData/Carthage-eisksmslyyxxiwabpizespmcgabn': You don’t have permission to save the file “Carthage-eisksmslyyxxiwabpizespmcgabn” in the folder “DerivedData”.

** CLEAN FAILED **

Couldn't set atribute on intermediates directory: No such file or directory (2)
Couldn't set atribute on intermediates directory: No such file or directory (2)
Couldn't set atribute on intermediates directory: No such file or directory (2)
make: *** [clean] Error 65
make: *** Waiting for unfinished jobs....

HOMEBREW_VERSION: 1.7.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: f239b7a62790cfb4f33bda9aab7597d43f1b1e3b
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: eeb085c4015bdbc79a608a910cfa195b59467705
Core tap last commit: 12 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CACHE: /Users/tjnickerson/Library/Caches/Homebrew
CPU: octa-core 64-bit kabylake
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1000
Git: 2.17.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 1.8.0_181
macOS: 10.14-x86_64
CLT: 10.0.0.0.1.1535735448
CLT headers: N/A
Xcode: 10.0
XQuartz: N/A

HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
MAKEFLAGS: -j8
CMAKE_PREFIX_PATH: /usr/local
CMAKE_INCLUDE_PATH: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/libxml2:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.14
HOMEBREW_GIT: git
HOMEBREW_SDKROOT: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Homebrew/Library/Homebrew/shims/mac/super:/usr/bin:/bin:/usr/sbin:/sbin
```

I've confirmed that the DerivedData path it's trying to access is owned by my user account, and belongs to the admin group. I even recursively added a 777 permission code to the directory, yet the error persists. I was able to install fine running the makefile from the repo locally, so I suspect it has to do with Homebrew sandboxing, though I'm not sure if this is a fix for them or for Carthage. I was wondering if anyone might have insight on how to resolve this.

As always, great work from the Carthage team :)

**Comment by mdiep** [comment]
Unfortunately, I have no idea what might be going on here! The homebrew installation is pretty far outside of my expertise.

**Comment by blender** [comment]
This seems to a be problem with you particular installation as carthage builds successfully on Mojave via homebrew. 

See https://github.com/Homebrew/homebrew-core/pull/32544 and https://github.com/Homebrew/homebrew-core/pull/32403 . Failures are not related to building Carthage itself.

**Comment by Nickersoft** [comment]
@blender Is this worth discussing with the Homebrew team? Literally, all I did was install Mojave and XCode 10 on a brand new Mac, run `brew install carthage` and got the above errors. I wound up being able to eventually install Carthage by just cloning the repo. I'm wondering if maybe it has to do with System Integrity Protection? I noticed the XCode build was throwing numerous "Operation Not Permitted" errors. I tried searching those errors, and ironically, this post was the fourth result in my Google search.
